clients/src/main/java/org/apache/kafka/clients/admin/DescribeConsumerGroupsResult.java

```diff
@@ -34,39 +35,40 @@
 @InterfaceStability.Evolving
 public class DescribeConsumerGroupsResult {
 
-    private final Map<String, KafkaFuture<ConsumerGroupDescription>> futures;
+    private final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures;
 
-    public DescribeConsumerGroupsResult(final Map<String, KafkaFuture<ConsumerGroupDescription>> futures) {
+    public DescribeConsumerGroupsResult(final Map<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> futures) {
         this.futures = futures;
     }
 
     /**
      * Return a map from group id to futures which yield group descriptions.
      */
     public Map<String, KafkaFuture<ConsumerGroupDescription>> describedGroups() {
-        return futures;
+        Map<String, KafkaFuture<ConsumerGroupDescription>> describedGroups = new HashMap<>();
+        for (Map.Entry<CoordinatorKey, KafkaFutureImpl<ConsumerGroupDescription>> entry : futures.entrySet()) {
+            describedGroups.put(entry.getKey().idValue, entry.getValue());
+        }
```

Comment:
Nit: Similar to how it's done in `DeleteConsumerGroupsResult`, perhaps we can replace this with a lambda expression instead?
```java
futures.forEach((key, future) -> describedGroups.put(key.idValue, future));
```